pages/_app.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import NextApp from 'next/app'
+
+import Layout from '../components/layout'
+
+class App extends NextApp {
+  static async getInitialProps({ Component, ctx }) {
+    let pageProps = {}
+
+    if (Component.getInitialProps)
+      pageProps = await Component.getInitialProps(ctx)
+
+    return { pageProps }
+  }
+
+  render() {
+    const { Component, pageProps } = this.props
+
+    return (
+      <Layout>
+        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
```

Comment:
I prefer spread rest props. It's just convenient.

When you write a wrapper around an element, let's say `<input>` you often do not even know it's attributes. Also, there is a wide range of global attributes such as id, lang, class, aria-* that a user might want to pass but component does not allow it. `restProps` is a way to provide usability and be safe it development. You extract what you want but pass everything else down to the child.